lib/ansible/plugins/action/bigip.py

```diff
@@ -0,0 +1,110 @@
+#
+# (c) 2016 Red Hat Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import sys
+import copy
+
+from ansible import constants as C
+from ansible.module_utils.basic import AnsibleFallbackNotFound
+from ansible.module_utils.f5_utils import F5_COMMON_ARGS
+from ansible.module_utils.six import iteritems
+from ansible.plugins.action.normal import ActionModule as _ActionModule
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class ActionModule(_ActionModule):
+
+    def run(self, tmp=None, task_vars=None):
+        if self._play_context.connection != 'local':
+            return dict(
+                failed=True,
+                msg='invalid connection specified, expected connection=local, got %s' % self._play_context.connection
```

Comment:
Is this going to cause existing 2.4 playbooks to fail?

Reply:
I'll have to defer to @caphrim007 on that one. My test environment doesn't allow me to validate the rest api, but I was originally told by @caphrim007 that the existing modules in 2.4 were not affected.

Reply:
@gundalow ahh, true.

F5 modules are not only used with connection local. They are also used with connection ssh and delegate_to. I think the above would fail if one used delegation.